### ISSUE TITLE
DL3042 fails despite --no-cache-dir being added

### ISSUE THREAD

**Issue by petarGitNik** [hadolint/hadolint#500]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Pass.

### Actual behavior

```
/dev/stdin:14 DL3042 Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`
```

### Steps to reproduce the behavior

Run:

```
docker run --rm -i hadolint/hadolint < ./Dockerfile
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.19.0-44-gf364c91
```

**Dockerfile** (if relevant)

```Dockerfile
FROM python:3.6.10-slim-buster

RUN pip install --no-cache-dir pipenv==2018.11.26 \
  && pipenv install requests \
  && rm -rf /root/.cache/pip /root/.cache/pipenv /root/.cache/pip-tools
```

It seems that `pipenv` is mistaken for pip. pipenv does not support `--no-cache-dir` flag.

**Additional environment details (OS, stack version, etc.)**

Used with dind: docker:18.09.5

**Comment by rodrigoaguilera** [comment]
This is a duplicate of https://github.com/hadolint/hadolint/issues/366
and therefore it should be closed
